Add absolute tolerance to KMeans algorithm

MassimoCimmino opened the pull request:
This closes #1985 and #1976.

I did not add any test since the issue occured in the already-implemented cases. Issue #1976 was caused by an error in the initialization of centroid positions.

mwetter commented:
@MassimoCimmino : Thanks for the changes. The verification fails due to a change in result of 0.4 K in one model, and in `IBPSA.Utilities.Clustering.Validation.KMeans_1d`, the returned clustering is different from the expected `labelsExp`. Can you please review and update as needed.

MassimoCimmino commented:
I updated `labelsExp` in `IBPSA.Utilities.Clustering.Validation.KMeans_1d`, `IBPSA.Utilities.Clustering.Validation.KMeans_2d` and `IBPSA.Fluid.Geothermal.Borefields.BaseClasses.HeatTransfer.ThermalResponseFactors.Validation.ClusterBoreholes_100boreholes`. The changes to `IBPSA.Utilities.Clustering.KMeans` enables the algorithm to find clusters with lower inertia. The difference in clusters in `IBPSA.Fluid.Geothermal.Borefields.BaseClasses.HeatTransfer.ThermalResponseFactors.Validation.ClusterBoreholes_100boreholes` would also explain the new results from the other models which use the same borefield geometry. I updated the results there also.
